bundles/ch.elexis.core.findings.util/src/ch/elexis/core/findings/util/FindingsTextUtil.java

```diff
@@ -70,111 +47,107 @@ public static String getGroupText(IObservation observation, boolean shouldSet) {
 		return stringBuilder.toString();
 	}
 
-	private static String getComponentText(ObservationComponent component, boolean includeUnit) {
-		StringBuilder stringBuilder = new StringBuilder();
-
-		ObservationType observationType = component.getTypeFromExtension(ObservationType.class);
-
-		if (ObservationType.TEXT.equals(observationType)) {
-			stringBuilder.append(component.getStringValue().orElse(StringUtils.EMPTY));
-		} else if (ObservationType.NUMERIC.equals(observationType)) {
-			try {
-				stringBuilder.append(
-						component.getNumericValue().isPresent() ? component.getNumericValue().get().toPlainString()
-								: StringUtils.EMPTY);
-				if (includeUnit) {
-					stringBuilder.append(StringUtils.SPACE);
-					stringBuilder.append(component.getNumericValueUnit().orElse(StringUtils.EMPTY));
-				}
-			} catch (NumberFormatException e) {
-				LoggerFactory.getLogger(FindingsTextUtil.class).warn("number illegal format", e);
-			}
-		}
-
-		return stringBuilder.toString();
+    public static String getObservationText(IObservation observation, boolean shouldSet) {
+        String title = getNameText(observation);
+        String value = getValueText(observation);
+
+        StringBuilder stringBuilder = new StringBuilder();
+        if (StringUtils.isNotBlank(title)) {
+            stringBuilder.append(title).append(" ");
+        }
+        stringBuilder.append(value);
+
+        if (shouldSet) {
+            observation.setText(stringBuilder.toString());
+        }
+
+        return stringBuilder.toString();
+    }
+
+	public static String getNameText(IObservation observation) {
+        return ModelUtil.getCodeBySystem(observation.getCoding(), CodingSystem.ELEXIS_LOCAL_CODESYSTEM)
+                .map(ICoding::getDisplay).orElse(StringUtils.EMPTY);
+    }
+
+	public static String getValueText(IObservation observation) {
+        switch (observation.getObservationType()) {
+            case TEXT:
+                return getTextValue(observation);
+            case NUMERIC:
+                return getNumericValue(observation);
+            case BOOLEAN:
+                return getBooleanValue(observation);
+            case DATE:
+                return getDateValue(observation);
+            case COMP:
+                return getCompValue(observation);
+            default:
+                LoggerFactory.getLogger(FindingsTextUtil.class).warn("Unknown ObservationType " + observation.getObservationType());
+                return StringUtils.EMPTY;
+        }
+    }
+
+	public static String getTextValue(IObservation observation) {
```

Comment:
Die Methode bitte in `getStringValue` umbenennen.

Bitte eine Methode `getCommentText` verwenden, und bei allen `getxxxValue` Methoden für den Kommentar Text verwenden.